### ISSUE TITLE
Great work!  when are you planning to release image-to-video models?

### ISSUE THREAD

**Issue by gxd1994** [THUDM/CogVideo#88]

**Comment by zRzRzRzRzRzRzR** [comment]
Thank you for your support. However, this might take some time as we currently do not have any related plans in the near future. Thank you for your understanding.

**Comment by StarCycle** [comment]
Hi @zRzRzRzRzRzRzR 

But as you mentioned in your paper, you already have an image-to-video version of CogVideoX

![图片](https://github.com/user-attachments/assets/192de65d-d980-4937-9233-f7620bb6240c)

**Comment by tengjiayan20** [comment]
> Hi @zRzRzRzRzRzRzR
> 
> But as you mentioned in your paper, you already have an image-to-video version of CogVideoX
> 
> ![图片](https://private-user-images.githubusercontent.com/33491471/355837737-192de65d-d980-4937-9233-f7620bb6240c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwNDAwMjcsIm5iZiI6MTcyMzAzOTcyNywicGF0aCI6Ii8zMzQ5MTQ3MS8zNTU4Mzc3MzctMTkyZGU2NWQtZDk4MC00OTM3LTkyMzMtZjc2MjBiYjYyNDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDE0MDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YTAzMzNiODBhZDdkNGQyMGJhOGE0NjIwMDNjZGY3ZGZmOGNiMjZmNzBjYjQyOWI0MzJhMzVkYTg5MDdiNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SpCRl7MK_NtyHe0sS_O5tKsLjOACZbCgyBoXip1yFDs)

Yes, the above reply means that we do not plan to open source the image-generated video model in the near future. Please pay attention and look forward to it.

**Comment by matbee-eth** [comment]
> > Hi @zRzRzRzRzRzRzR
> > But as you mentioned in your paper, you already have an image-to-video version of CogVideoX
> > ![图片](https://private-user-images.githubusercontent.com/33491471/355837737-192de65d-d980-4937-9233-f7620bb6240c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwNDAwMjcsIm5iZiI6MTcyMzAzOTcyNywicGF0aCI6Ii8zMzQ5MTQ3MS8zNTU4Mzc3MzctMTkyZGU2NWQtZDk4MC00OTM3LTkyMzMtZjc2MjBiYjYyNDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDE0MDg0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YTAzMzNiODBhZDdkNGQyMGJhOGE0NjIwMDNjZGY3ZGZmOGNiMjZmNzBjYjQyOWI0MzJhMzVkYTg5MDdiNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SpCRl7MK_NtyHe0sS_O5tKsLjOACZbCgyBoXip1yFDs)
> 
> Yes, the above reply means that we do not plan to open source the image-generated video model in the near future. Please pay attention and look forward to it.

Would absolutely appreciate the release of the img+text to video :)

**Comment by StarCycle** [comment]
Hi @tengjiayan20,

Thank you for the response! 

Is it difficult to finetune an image-to-video model by myself on the [WebVid10M](https://huggingface.co/datasets/TempoFunk/webvid-10M) dataset? How many samples and trainning steps do you need to do that?

Do you apply a fixed noise level on the image condition in the diffusion process?

Sorry but I really need a image2video model in my application.

Best wishes,
StarCycle

**Comment by tengjiayan20** [comment]
> Hi @tengjiayan20,
> 
> Thank you for the response!
> 
> Is it difficult to finetune an image-to-video model by myself on the [WebVid10M](https://huggingface.co/datasets/TempoFunk/webvid-10M) dataset? How many samples and trainning steps do you need to do that?
> 
> Do you apply a fixed noise level on the image condition in the diffusion process?
> 
> Sorry but I really need a image2video model in my application.
> 
> Best wishes, StarCycle

1. I think it is ok. After all, many image-to-video works have verified that webvid dataset can satisfy this task. The key is that they don't have a better base text-to-video model.
2. Augmentation in training is beneficial

**Comment by eugenelyj** [comment]
@tengjiayan20

**Comment by tengjiayan20** [comment]
> @tengjiayan20 Dear author, Do you apply the same noise level for all the timestep training, or do you apply timestep-dependent noise adding when training the image-to-video model?

Usually the strength of augmentation is random and dynamic. Since the augmentation is added to the image condition, and the image condition is constant during sampling and does not change with the timestep, I think the strength of augmentation does not need to change with the timestep. It is just to enhance robustness to fill the gap between the conditions during training and inference.

**Comment by jinhuaca** [comment]
Do you do conditioning similarly in i2v models as compared to t2v models? For example, do you concatenate the image embeddings (instead of text embeddings) with the video tokens as conditioning? Or instead, do you replace the first frame of the latent model with the image?

**Comment by trouble-maker007** [comment]
@tengjiayan20 Could you please share a general training method for image to video, such as how to inject image conditions into the model? concat the image embedding with the text embedding or just replace the noise latent with the image embedding?

**Comment by eugenelyj** [comment]
@tengjiayan20 Dear author, one more question....Do you joint training the text-to-image when you fine-tune the image-to-video? Because for image-to-video, the latent channel have been doubled (concat with first frame), I am confused how to do it.

**Comment by wangqiang9** [comment]
> @tengjiayan20 Dear author, one more question....Do you joint training the text-to-image when you fine-tune the image-to-video? Because for image-to-video, the latent channel have been doubled (concat with first frame), I am confused how to do it.

SVD use a conv to upscale the doubled channels...

**Comment by eugenelyj** [comment]
@zRzRzRzRzRzRzR Dear author, can you share the implementation of `add_noise_to_first_frame`, in particular the detailed parameters (distribution) of the added noise.

**Comment by zRzRzRzRzRzRzR** [comment]
yes,check our sat code now, is in pr

**Comment by zRzRzRzRzRzRzR** [comment]
And I2V model will opensource in next 24 hours, this issue close
